### PR TITLE
insert isTRUE in calc/sim/calcDiff to avoid uncompiled errors with truncation/dyn idx

### PR DESCRIPTION
**Pull request by paciorek**
FRO-DNM

This inserts isTRUE in calculate/simulate/calculateDiff nodefunctions so that under truncation (if a node value is NA) or dyn idx (if an index is NA) users don't see untrapped R errors from `if(NA)`. 

This also creates an `isTRUE` identity function in our C++ code. Presumably this will have negligible effect on compiled run times but worth discussion.

Before merging, check line 17 of nimbleFunction_nodeFunction to see why there is an unexpected change relative to devel.

### CODE REVIEW

**Comment by paciorek** [comment]
The line 17 change was inexplicable, but I've fixed it so we use `nimRep` not `nimArray`.

**Comment by paciorek** [comment]
@perrydv  can you take a quick look here? C++ changes are trivial, but just want to make sure I'm not missing something.

File: packages/nimble/R/nimbleFunction_nodeFunction.R [hunk 1]
```diff
@@ -272,17 +272,17 @@ ndf_createStochCalculateTrunc <- function(logProbNodeExpr, LHS, RHS, diff = FALS
                                                )), list( e = substCode)))
         } else {
             if(discrete && lower != -Inf) {
-                substCode <- quote(if(CONDITION) {
-                                       if(LOWER <= VALUE & VALUE <= UPPER)
+                substCode <- quote(if(TRUE(CONDITION)) {
```
Comment (perrydv) [inline packages/nimble/R/nimbleFunction_nodeFunction.R]: @paciorek Is this one a typo?

File: packages/nimble/inst/include/nimble/Utils.h [hunk 2]
```diff
@@ -165,6 +165,8 @@ void nimStop();
 
 bool nimNot(bool x);
 
+bool isTRUE(bool x);
```
Comment (perrydv) [inline packages/nimble/inst/include/nimble/Utils.h]: @paciorek I suggest in the .h:
static inline bool isTRUE(bool x) {return(x);}
and then nothing is needed in the .cpp and the compiler should inline the call.

**Comment by paciorek** [comment]
Waiting on Travis; then will merge.

**Pull request merged**